collect_app/src/main/java/org/odk/collect/android/utilities/ToastUtils.java

```diff
@@ -34,4 +37,29 @@ private static void showToast(String message, int duration) {
     private static void showToast(int messageResource, int duration) {
         Toast.makeText(Collect.getInstance(), Collect.getInstance().getString(messageResource), duration).show();
     }
+
+    @SuppressLint("ShowToast")
+    public static void showShortToastInMiddle(int messageResource) {
+        showToastInMiddle(Toast.makeText(Collect.getInstance(), Collect.getInstance().getString(messageResource), Toast.LENGTH_SHORT));
+    }
+
+    @SuppressLint("ShowToast")
+    public static void showShortToastInMiddle(String message) {
+        showToastInMiddle(Toast.makeText(Collect.getInstance(), message, Toast.LENGTH_SHORT));
+    }
+
+    @SuppressLint("ShowToast")
+    public static void showLongToastInMiddle(int messageResource) {
+        showToastInMiddle(Toast.makeText(Collect.getInstance(), Collect.getInstance().getString(messageResource), Toast.LENGTH_LONG));
+    }
+
+    private static void showToastInMiddle(Toast toast) {
```

Comment:
Getting a `Toast` in here is unexpected. I was suggesting 

```
    public static void showShortToastInMiddle(int messageResource) {
        showToastInMiddle(Collect.getInstance().getString(messageResource), Toast.LENGTH_SHORT);
    }

    public static void showShortToastInMiddle(String message) {
        showToastInMiddle(message, Toast.LENGTH_SHORT);
    }

    public static void showLongToastInMiddle(int messageResource) {
        showToastInMiddle(Collect.getInstance().getString(messageResource), Toast.LENGTH_LONG);
    }

    private static void showToastInMiddle(String message, int duration) {
        Toast toast = Toast.makeText(Collect.getInstance(), message, duration);
        ViewGroup group = (ViewGroup) toast.getView();
        TextView messageTextView = (TextView) group.getChildAt(0);
        messageTextView.setTextSize(21);
        messageTextView.setGravity(Gravity.CENTER);

        toast.setGravity(Gravity.CENTER, 0, 0);
        toast.show();
    }
```

That way you don't need to suppress a lint check and the method name better matches what it does.

Reply:
Ok I see. Thanks for your explanation!